### DISCUSSION TITLE
Volume Meter

### DISCUSSION THREAD

**Discussion by madhud3** [jspsych/jsPsych#2198] [Q&A]
Hello I am trying to make a study with audio files and we need to have a volume test before the experiment.  I found this code but I am having trouble implementing it into my experimental code: 

https://ourcodeworld.com/articles/read/413/how-to-create-a-volume-meter-measure-the-sound-level-in-the-browser-with-javascript

Has anyone tried this or have a way to help as some of the code seems to be deprecated.

**Comment by becky-gilbert** [comment]
Hi @madhud3, just to clarify what you want to do: you want to request access to the user's microphone, play audio files, pick up that audio output through the user's mic, and display the volume levels from the mic stream on the screen. Is that right? Do you need to save any information about the volume test to the data, or is this just meant to be a visual indicator for the participant?

I haven't done this but might be interested in trying it, because I am trying to figure out a way to sync the timing from an audio recording with the onset of a stimulus. One way to do this would be to play a tone that is synchronized with the visual display, so that the audio recording contains a tone that marks the visual stimulus onset as well as the participant's spoken response.

Also, if I've understood correctly, one thing that occurs to me is that the levels that are generated using this method would depend on both the level/quality of the audio coming out of the computer (which I think is what you're interested in?), as well as the quality and positioning of the microphone that's picking up the audio. So that might be a problem if you're trying to compare volume level values across different participants.

**Reply by madhud3** [reply]
Hello @becky-gilbert thanks so much for your comment. So we are doing a false memory experiment and presenting words auditorily to participants. we want to make sure they have their volume at an appropriate level in order to hear all the words being presented without having to adjust the volume. Therefore, we want a volume test before the start of the experiment just to make sure they know they have their volume at an appropriate level. Maybe it would be good to then take each participants volume level (0-100) so we can have that data, but it is not necessary for us to take their volume. we just want this for the participants to know they will be able to hear properly throughout the experiment. I hope this makes sense.

**Reply by jodeleeuw** [reply]
Hi @madhud3,

This is a different approach, but one thing that I've done when I want to verify that participants can hear audio in my experiment is to move some of the instructions to audio. For example, you can set up a trial with the `audio-keyboard-response` plugin where the audio file says something like "Please adjust your volume to a comfortable level and then press the Q key to continue", while setting the `choices` parameter to only accept Q as a valid response. You can use the `prompt` parameter to display a message like "Please follow the audio instructions to continue." If you do this, you'll probably want to loop the audio for an extended time to give participants a chance to adjust.

**Reply by becky-gilbert** [reply]
Yes, I was going to suggest something similar to what Josh said, as this would be much easier to set up. What I've done in my experiments is used one of the `audio-*` plugins to display a `prompt` asking the participant to adjust the volume to a comfortable level, and then type the sentence that they hear into a text box. The audio stimulus is a spoken sentence that is either played on a loop (as Josh suggests) or has the playback controls on the screen so that the participant can re-play it as many times as they'd like. Unlike Josh's suggestion, this one would require modifications to the plugin (to display a textbox, and possibly the audio playback controls), but these modifications would be _much_ easier than using the mic to show audio volume levels.
